src/main/java/rx/internal/operators/OperatorDistinctUntilChanged.java

```diff
@@ -68,7 +83,16 @@ public void onNext(T t) {
                 previousKey = key;
                 
                 if (hasPrevious) {
-                    if (!(currentKey == key || (key != null && key.equals(currentKey)))) {
+                    Boolean comparison;
+                    
+                    try {
+                        comparison = comparator.call(currentKey, key);
+                    } catch (Throwable e) {
+                        Exceptions.throwOrReport(e, child, key);
+                        return;
+                    }
+                    
+                    if (!comparison) {
```

Comment:
Good catch! I'll change the comparison to be primitive boolean to trigger NPE inside the try-block if necessary.